scipy/optimize/constraints.py

```diff
@@ -0,0 +1,617 @@
+from __future__ import division, print_function, absolute_import
+import numpy as np
+import scipy.sparse as spc
+
+
+__all__ = ['CanonicalConstraint',
+           'NonlinearConstraint',
+           'LinearConstraint',
+           'BoxConstraint',
+           'check_sparsity',
```

Comment:
This is a utility function, you don't mean to expose this as ``optimize.check_sparsity`` right? True for several other functions here as well I suspect, however you do an ``import *`` from ``optimize/__init__.py``.